pennylane/templates/subroutines/select.py

```diff
@@ -98,6 +99,21 @@ def __init__(self, ops, control, id=None):
         all_wires = target_wires + control
         super().__init__(*self.data, wires=all_wires, id=id)
 
+    def __copy__(self):
+        """Copy this op"""
+        cls = self.__class__
+        copied_op = cls.__new__(cls)
+
+        new_data = copy.copy(self.data)
+
+        for attr, value in vars(self).items():
+            if attr != "data":
+                setattr(copied_op, attr, value)
+
+        copied_op.data = new_data
```

Comment:
Couldn't this be done within the above `for` loop with `else` 🤔

Reply:
It could. But I don't think that's necessary. Would it make the code cleaner in your opinion?

Reply:
It's cleaner either way. I was just confirming.

Reply:
Oh. Actually, maybe it can't. If we put this line inside the loop, we may call `copied_op.data = new_data` before `copied_op.hyperparameters` is initialized. That's what causes the error we are fixing.